Medical-imaging example, Test an image error

Hello, i have problem with Medical-imaging example.

"Test a record from valiation set" works fine, but when i try to put path in "Test an image"

Which file types i can use there dcm or png?

![1iss](https://cloud.githubusercontent.com/assets/20860578/19886036/1d6e935e-a03b-11e6-9d17-39aa6673e297.png)

i have an error in both cases(dcm/png),( the path is correct 100%)

```
2016-11-01 13:56:04 [ERROR] Error: None: No such file or directory
Traceback (most recent call last):
File "/digits/MS/plugins/DIGITS/digits/tools/inference.py", line 293, in <module>
args['resize']
File "/digits/MS/plugins/DIGITS/digits/tools/inference.py", line 93, in infer
reader = DbReader(input_list)
File "/digits/MS/plugins/DIGITS/digits/utils/lmdbreader.py", line 21, in __init__
lock=False)
lmdb.Error: None: No such file or directory
```
Can anyone help me with this?

--- a/plugins/data/sunnybrook/digitsDataPluginSunnybrook/data.py
+++ b/plugins/data/sunnybrook/digitsDataPluginSunnybrook/data.py
@@ -87,16 +87,16 @@ def get_all_contours(contour_path):
 def load_contour(contour, img_path):
     filename = "IM-%s-%04d.dcm" % (SAX_SERIES[contour.case], contour.img_no)
     full_path = os.path.join(img_path, contour.case, filename)
-    f = dicom.read_file(full_path)
-    img = f.pixel_array.astype(np.int)
+    img = load_image(full_path)
     ctrs = np.loadtxt(contour.ctr_path, delimiter=" ").astype(np.int)
     label = np.zeros_like(img, dtype="uint8")
     cv2.fillPoly(label, [ctrs], 1)
     return img, label
 
-#
-# Main class
-#
+
+def load_image(full_path):
+    f = dicom.read_file(full_path)
+    return f.pixel_array.astype(np.int)
 
 
 @subclass
@@ -130,7 +130,12 @@ class DataIngestion(DataIngestionInterface):
 
     @override
     def encode_entry(self, entry):
-        img, label = load_contour(entry, self.image_folder)
+        if isinstance(entry, basestring):
+            img = load_image(entry)
+            label = np.array([0])
+        else:
+            img, label = load_contour(entry, self.image_folder)
+            label = label[np.newaxis, ...]
 
         if self.userdata['channel_conversion'] == 'L':
             feature = img[np.newaxis, ...]
@@ -140,8 +145,6 @@ class DataIngestion(DataIngestionInterface):
             feature[0] = img
             feature[1] = img
             feature[2] = img
-
-        label = label[np.newaxis, ...]
 
         return feature, label
 
@@ -213,6 +216,14 @@ class DataIngestion(DataIngestionInterface):
                 entries = ctrs[:n_val_entries]
         elif stage == constants.TEST_DB:
             if self.userdata['validation_record'] != 'none':
+                if self.userdata['test_image_file']:
+                    raise ValueError("Specify either an image or a record "
+                                     "from the validation set.")
+                # test record from validation set
                 entries = [ctrs[int(self.validation_record)]]
+
+            else:
+                # test image file
+                entries = [self.userdata['test_image_file']]
 
         return entries

--- a/plugins/data/sunnybrook/digitsDataPluginSunnybrook/forms.py
+++ b/plugins/data/sunnybrook/digitsDataPluginSunnybrook/forms.py
@@ -86,7 +86,7 @@ class InferenceForm(Form):
         validators=[
             validate_file_path,
         ],
-        tooltip="Provide an image"
+        tooltip="Provide the (server) path to an image."
     )
 
     validation_record = utils.forms.SelectField(

--- a/plugins/data/sunnybrook/digitsDataPluginSunnybrook/templates/inference_template.html
+++ b/plugins/data/sunnybrook/digitsDataPluginSunnybrook/templates/inference_template.html
@@ -11,7 +11,7 @@
             <div class="form-group{{mark_errors([form.test_image_file])}}">
                 {{ form.test_image_file.label }}
                 {{ form.test_image_file.tooltip }}
-                {{ form.test_image_file(class='form-control', placeholder='image file') }}
+                {{ form.test_image_file(class='form-control autocomplete_path', placeholder='image file') }}
             </div>
         </div>
     </div>

